src/types.ts

```diff
@@ -34,8 +34,8 @@ export interface BalanceResponse {
 }
 
 export interface DripResponse {
```

Comment:
it makes more sense to model this as a union type

`export type DripResponse = Error | { hash: string }`

That way you would be able to handle it more safely in other places

Reply:
What do you mean?

Reply:
Dealing with this DripResponse type is not good. Example from the code:

```typescript
      const message = res.data.hash
        ? `Sent ${sender} ${dripAmount} ${unit}s. Extrinsic hash: ${res.data.hash}`
        : (res.data.error || 'An unexpected error occured, please check the server logs');
```

Since it has two optional fields, it might lead to bugs where neither is defined. Also handling it is inconsistent: here you are **choosing** for `data.hash` and, if it does not exist, checking for `data.error`; however, there's no reason why it could not be `data.error` first and `data.hash` after.

When a DripResponse has `{ hash: "something", error: "something" }`, what does it mean? Was it an error or success? Really, it should never be in two states ("error" and "success") at the same time, but it's currently allowed because the error and success information are encoded in the same type.

If it's defined as distinct types, handling it would be safer and saner:

```typescript
      const message = res instanceof Error
        ? res.message
        :  `Sent ${sender} ${dripAmount} ${unit}s. Extrinsic hash: ${res.data.hash}`
```

If you want to apply this change, I suggest replacing the type with the signature I provided and start fixing the errors which appear.

Reply:
Fair enough. Firstly it was separated like `{hash} | {error}`.
But I can't user Error directly, because it throws.